File: MySite.Shared/Models/About/Experience.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MySite.Shared.Models.About
{
    public class Experience
    {
        public long ExperienceId { get; set; }

        public virtual WorkPosition Position { get; set; }

        public virtual Company Company { get; set; }

        public DateTime WorkBegin { get; set; }

        /// <summary>
        /// If this property is not set it means that i'm currently working there.
        /// </summary>
        public DateTime? WorkEnd { get; set; }

        public virtual Localization Localization { get; set; }
    }
}

File: MySite.Client/ClientApp/src/app/models/experience.ts
import { Localization } from "./localization";
import { Company } from "./company";
import { WorkPosition } from "./work-position";
import { LocalizationInformation } from "./localization-information";

export interface Experience {
  id: number;
  positionId: number;
  position: WorkPosition;
  companyId: number;
  company: Company;
  workBegin: Date;
  workEnd?: Date;
  localizationId: number;
  localization: Localization;
}
